src/battle/star/chill_out_7900D0.c

```diff
@@ -2,20 +2,81 @@
 
 extern s32 D_802A2CC0;
 
-INCLUDE_ASM(s32, "battle/star/chill_out_7900D0", func_802A1000_7900D0);
+ApiStatus func_802A1000_7900D0(ScriptInstance* script, s32 isInitialCall) {
+    Bytecode* args = script->ptrReadPos;
+    s32 a = get_variable(script, *args++);
+    s32 b = get_variable(script, *args++);
+    s32 c = get_variable(script, *args++);
 
-INCLUDE_ASM(s32, "battle/star/chill_out_7900D0", func_802A10AC_79017C);
+    func_8006FEF0(0, a, b, c, 40.0f);
+    return ApiStatus_DONE2;
+}
+
+ApiStatus func_802A10AC_79017C(ScriptInstance* script, s32 isInitialCall) {
+    Bytecode* args = script->ptrReadPos;
+    s32 a = get_variable(script, *args++);
+    s32 b = get_variable(script, *args++);
+    s32 c = get_variable(script, *args++);
+
+    func_80070550(5, a, b, c, 50.0f, 50.0f, 40, 30);
+    return ApiStatus_DONE2;
+}
 
-INCLUDE_ASM(s32, "battle/star/chill_out_7900D0", func_802A116C_79023C);
+ApiStatus func_802A116C_79023C(ScriptInstance* script, s32 isInitialCall) {
+    Bytecode* args = script->ptrReadPos;
+    s32 a = get_variable(script, *args++);
+    s32 b = get_variable(script, *args++);
+    s32 c = get_variable(script, *args++);
+
+    func_8006FEF0(0, a, b, c, 30.0f);
+    return ApiStatus_DONE2;
+}
 
 INCLUDE_ASM(s32, "battle/star/chill_out_7900D0", func_802A1218_7902E8);
 
 #define NAMESPACE battle_star_chill_out
 #include "common/FadeBackgroundToBlack.inc.c"
 
-INCLUDE_ASM(s32, "battle/star/chill_out_7900D0", func_802A137C_79044C);
+ApiStatus func_802A137C_79044C(ScriptInstance* script, s32 isInitialCall) {
+    s32 temp;
+
+    if (isInitialCall) {
+        func_8011D82C(1);
+        *D_801512F0 = 1;
+        set_background_color_blend(0, 0, 0, 0);
+        script->functionTemp[0].s = 0;
+    }
+
+    set_background_color_blend(0, 0, 0, script->functionTemp[0].s);
+    script->functionTemp[0].s += 10;
+    temp = script->functionTemp[0].s;
+
+    if (temp >= 230) {
+        set_background_color_blend(0, 0, 0, 230);
+        return ApiStatus_DONE2;
+    }
 
-INCLUDE_ASM(s32, "battle/star/chill_out_7900D0", func_802A1414_7904E4);
+    return ApiStatus_BLOCK;
+}
+
+ApiStatus func_802A1414_7904E4(ScriptInstance* script, s32 isInitialCall) {
+    s32 temp;
+
+    if (isInitialCall) {
+        script->functionTemp[0].s = 20;
+    }
+
+    set_background_color_blend(0, 0, 0, script->functionTemp[0].s * 10);
+    script->functionTemp[0].s -= 1;
+    temp = script->functionTemp[0].s;
+
+    if (!temp) {
```

Comment:
Same here - I'm not sure `temp` is required. Also, since this isn't a boolean (rather, its an int counting down), it's probably better to leave the condition as `if (script->functionTemp[0].s == 0)`.